src/Filters/design.jl

```diff
@@ -234,61 +234,84 @@ Elliptic(n::Integer, rp::Real, rs::Real) = Elliptic(Float64, n, rp, rs)
 # returns frequency in half-cycles per sample ∈ (0, 1)
 function normalize_freq(w::Real, fs::Real)
     w <= 0 && throw(DomainError(w, "frequencies must be positive"))
-    f = 2*w/fs
+    f = 2 * w / fs
     f >= 1 && throw(DomainError(f, "frequencies must be less than the Nyquist frequency $(fs/2)"))
     f
 end
-
-struct Lowpass{T} <: FilterType
-    w::T
+function normalize_complex_freq(w::Real, fs::Real)
+    f = 2 * w / fs
+    f >= 2 && throw(DomainError(f, "frequencies must be less than the Nyquist frequency $(fs)"))
+    f
 end
 
 """
     Lowpass(Wn)
 
 Low pass filter with cutoff frequency `Wn`.
 """
-Lowpass(w::Real) = Lowpass{typeof(w/1)}(w)
-
-struct Highpass{T} <: FilterType
+struct Lowpass{T<:Real} <: FilterType
     w::T
+    Lowpass{T}(w::Real) where {T<:Real} = new{T}(w)
+    Lowpass(w::Real) = Lowpass{typeof(w / 1)}(w)
 end
```

Comment:
The diff is a bit messy, so maybe I'm missing something, but why can't this just be
```suggestion
struct Lowpass{T<:Real} <: FilterType
    w::T
end
Lowpass(w::Real) = Lowpass{typeof(w / 1)}(w)
```
(which, if I read correctly, doesn't change much, just adds the `<: Real` restriction on `T`)

Reply:
Ah, right, without any explicit inner constructors, a `Lowpass(w::Real)` method is auto-generated. (Before, a `Lowpass(w::Any)` was auto-generated which didn't conflict with the explicit `Lowpass(w::Real)`  being defined.) Somewhat unfortunate code-churn, but seems unavoidable then.

But while at it, what do think about:
```suggestion
struct Lowpass{T<:Real} <: FilterType
    w::T
    Lowpass{T}(w::Real) where {T<:Real} = new{T}(w)
    Lowpass(w::T) where {T<:Real} = Lowpass{float(T)}(w)
end
```
Might a bit more efficient for non-trivial types like `BigInt` or `BigFloat`, but I don't know whether it might make a semantic difference.

Reply:
Admitted, I haven't looked where we use instances of these types. What do we expect of the type of `w`?

Reply:
Or, to put it differently, what goes wrong with just:
```suggestion
struct Lowpass{T<:Real} <: FilterType
    w::T
end
```

Reply:
Then let's just leave this as `typeof(w / 1)` for now; mostly unrelated to the intention of the PR anyway.